eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/BundleBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Box;
+import com.schedjoules.eventdiscovery.framework.serialization.core.FluentBuilder;
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.Present;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class BundleBuilder implements FluentBuilder<Bundle>
+{
+    private final Bundle mBundle;
+
+
+    public BundleBuilder(Bundle bundle)
+    {
+        mBundle = bundle;
+    }
+
+
+    public BundleBuilder()
+    {
+        this(new Bundle());
+    }
+
+
+    @Override
+    public Bundle build()
+    {
+        return mBundle;
+    }
+
+
+    @Override
+    public <T> FluentBuilder<Bundle> withParam(Key<T> key, Box<T> box)
+    {
+        return withParam(key, new Present<>(box));
+    }
+
+
+    @Override
+    public <T> FluentBuilder<Bundle> withParam(Key<T> key, Optional<Box<T>> optBox)
+    {
+        Bundle bundle = new Bundle(mBundle);
```

Comment:
That's the beauty and the whole point of immutable objects. If we don't change anything we don't have to return a new instance and in fact there is no risk in returning the same instance and no benefit in returning a new instance. For the caller it doesn't matter at all.

Even if we use the same builder for multiple Bundles or Intents, this won't cause any problems, because the object is immutable. Any call to `with` that would change the state will create a new instance.